### ISSUE TITLE
Mutable version of graph.edges

### ISSUE THREAD

**Issue by curiousdannii** [petgraph/petgraph#408]
### Summary

Can a mutable version of graph.edges be added?

### Motivation

If you want to iterate the edges of a node and then modify one of them you can't currently. Instead you need to keep track of the edge IDs to modify separately, and then modify them once you're finished iterating. If it was possible to modify them during the iteration that would be so much cleaner.

**Comment by bluss** [comment]
You can currently use a detached neighbors iterator to do this

**Comment by curiousdannii** [comment]
Ah yes, that does look like it could work for me. I hadn't thought of it because it's only available as `.neighbours(n).detach()` not `.edges(n).detach()`.

**Comment by curiousdannii** [comment]
I've started using `.neighbours(n).detach()` but I've run into some issues. I think they ultimately stem from the fact that there's no trait for it, so I have to use the type `Graph<N, E, Ty, Ix>` in my functions instead of a trait, which then makes lots of other things more complex.

Would it be possible to move it to its own trait?

**Comment by bluss** [comment]
If you find a way to traitify it, it would be good to not move, but provide a graph trait that has this functionality. Note that petgraph is a bit short on maintenance resources.
